aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1092,6 +1092,77 @@ Tensor permute(const Tensor& self, IntArrayRef dims) {
   return self.as_strided(newSizes, newStrides);
 }
 
+Tensor permute_sparse_coo(const Tensor& self, IntArrayRef dims) {
+  const auto ndim = self.dim();
```

Comment:
Does it make sense to create a helper function so you can share these error messages with `permute` above? That way they're both consistent.

Reply:
It could be done, but the variables created during these checks are needed in both implementations.